### PR TITLE
Remove uses of TimeStepId in interpolation

### PR DESCRIPTION
**Pull request by wthrowe**
TimeStepId is not a global identifier, so it cannot be used as an ID
for the interpolator component.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp
Comment (wthrowe) [inline src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:66]: @markscheel What are these two functions?  Do we need both of them?

**Comment by wthrowe** [comment]
There are some more of these that I missed.

**Comment by wthrowe** [comment]
Got the rest.

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.cpp
Comment (nilsvu) [inline src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.cpp:6]: Nice, I believe this removes the dependence of `ParallelInterpolation` on `Time`. Could you double-check, and then remove the dependency from the `CMakeLists.txt`?

Comment (wthrowe) [reply src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.cpp:6]: Yep, looks like that's the last one for the main library.  The test library still uses it, though.  Squashed directly.

**Review by nilsvu — APPROVED** [review]

**Comment by wthrowe** [comment]
@markscheel ping

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp
Comment (markscheel) [reply src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:66]: These two functions were added by @moxcodes in #3421 so that certain code would work both for `double` and for `TimeStepId`, without writing two different versions of that code.

Jordan needed temporal_id to be something different than a double so that self-start would work with #2323 .

If the temporal_id is now always a double, then these two functions should be removed.

File: tests/Unit/ParallelAlgorithms/Interpolation/Test_CleanUpInterpolator.cpp
Comment (markscheel) [inline tests/Unit/ParallelAlgorithms/Interpolation/Test_CleanUpInterpolator.cpp:38]: Why is this called `OtherId`? Why not `TimeId` or something that has `Time` in it?

File: tests/Unit/ParallelAlgorithms/Interpolation/Test_InitializeInterpolator.cpp
Comment (markscheel) [inline tests/Unit/ParallelAlgorithms/Interpolation/Test_InitializeInterpolator.cpp:29]: `TimeId`? `TimeTag`?

**Review by markscheel — COMMENTED** [review]
A minor comment about a variable name.

My biggest concern is that in #3421 @moxcodes reverted a PR similar to this one because he needed interpolation during the self-start of GH+CCE, and he couldn't use `double` as a time identifier.  Has that changed?  And how does this PR impact #2323 ?

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp
Comment (wthrowe) [reply src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:66]: Maybe we should hold off until #3422 is merged so we can see if there's any interaction?

File: tests/Unit/ParallelAlgorithms/Interpolation/Test_CleanUpInterpolator.cpp
Comment (wthrowe) [reply tests/Unit/ParallelAlgorithms/Interpolation/Test_CleanUpInterpolator.cpp:38]: The one it is "other" to is called "Time", so using "Time" and "TimeId" as the two IDs seemed rather confusing.  I could go with "Time2" or something?

**Review by wthrowe — COMMENTED** [review]
Yes, it looks like some of this may still be needed.  I really want to get rid of the non-CCE stuff, though, since using it that way is wrong and it keeps getting copy-pasted to more places.

File: tests/Unit/ParallelAlgorithms/Interpolation/Test_CleanUpInterpolator.cpp
Comment (markscheel) [reply tests/Unit/ParallelAlgorithms/Interpolation/Test_CleanUpInterpolator.cpp:38]: Looking at the code again, I believe that the only reason that there are 2 tags in this test is to test that interpolation works with both `TimeStepId` and `double`.  So if interpolation only works with `double`, then I think there doesn't need to be a 2nd tag at all.

**Comment by markscheel** [comment]
Currently, I think the idea is that you are supposed to use doubles instead of TimeStepIds everywhere except in CCE-GH self-start (and interpolation currently works with both doubles and TimeStepIds).  But yes it is annoying that people keep using TimeStepIds in most places (probably my fault, because the original version of interpolation used only TimeStepIds and not doubles).

**Comment by wthrowe** [comment]
In addition to needing to figure out CCE, this is on hold until the interactions between the control system and AH finder have been worked out to avoid merge conflicts.

**Comment by wthrowe** [comment]
This would need to be done very differently to work with CCE stuff.  I may revisit this at some point, but I think it will be as easy to start from scratch.

**Pull request closed without merging**
